app/main.py

```diff
@@ -1,23 +1,35 @@
 import init_django_orm  # noqa: F401
 
-from django.db.models import QuerySet
+from django.db.models import QuerySet, Q, F
+
+from db.models import LoyaltyProgram, LoyaltyProgramParticipant, Customer
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    return LoyaltyProgram.objects.values_list("name", "bonus_percentage")
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.filter(
+        last_activity__year=2021
+    ).values("customer__first_name")
 
 
 def most_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.order_by(
+        "-sum_of_spent_money"
+    ).values_list(
+        "customer__first_name", "customer__last_name", "sum_of_spent_money"
+    )[:5]
 
 
 def clients_with_i_and_o() -> QuerySet:
-    pass
+    return Customer.objects.filter(
+        Q(first_name__startswith="I") | Q(last_name__icontains="o")
```

Comment:
According to the task description, second check should be case sensitive.

Reply:
Fixed